staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -252,17 +250,34 @@ func (c *delegatingCRConverter) ConvertToVersion(in runtime.Object, target runti
 		return converted, nil
 	}
 
-	convertedObjects, err := c.converter.Convert(list, toGVK.GroupVersion())
+	// Deepcopy ObjectMeta because the converter might mutate the objects, and we
+	// need the original ObjectMeta furthermore for validation and restoration.
+	for i := range objectsToConvert {
+		original := objectsToConvert[i].Object
+		objectsToConvert[i].Object = make(map[string]interface{}, len(original))
```

Comment:
similar questions about whether modifying the list input in-place is ok

Reply:
same answer: "Note that the function may mutate in object and return it."